src/applications/caregivers/components/FormFields/AddressWithAutofill.jsx

```diff
@@ -82,10 +94,23 @@ const PrimaryAddressWithAutofill = props => {
   // define our non-checkbox input blur event
   const handleBlur = useCallback(
     event => {
-      const fieldName = event.target.name.split('_').pop();
+      const { name } = event.target;
+
+      if (!name) {
+        // blurring the County's AddressCountyDescription component causes
+        // an error below when attempting the split. this avoids cascading the
+        // remaining blur behavior to it.
+        return;
```

Comment:
🤷 This prevents the log message complaining about the `_undefined_.split` failure. I'm not sure if it's possible to cover this with our tests without `export`ing the `handleBlur` method or something else awkward? The failure doesn't cause a proper UI error or anything. Open to suggestions!

Reply:
Note this bug does manifest as a full-screen error, but only on localhost:

```
Uncaught runtime errors:
×
ERROR
Cannot read properties of undefined (reading 'split')
TypeError: Cannot read properties of undefined (reading 'split')
    at http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:62879:26
    at HTMLUnknownElement.callCallback (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164303:14)
    at Object.invokeGuardedCallbackDev (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164352:16)
    at invokeGuardedCallback (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164414:31)
    at invokeGuardedCallbackAndCatchFirstError (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164428:25)
    at executeDispatch (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168601:3)
    at processDispatchQueueItemsInOrder (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168633:7)
    at processDispatchQueue (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168646:5)
    at dispatchEventsForPlugins (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168657:3)
    at http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168866:12
ERROR
Cannot read properties of undefined (reading 'split')
TypeError: Cannot read properties of undefined (reading 'split')
    at http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:62879:26
    at HTMLUnknownElement.callCallback (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164303:14)
    at Object.invokeGuardedCallbackDev (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164352:16)
    at invokeGuardedCallback (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164414:31)
    at invokeGuardedCallbackAndCatchFirstError (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164428:25)
    at executeDispatch (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168601:3)
    at processDispatchQueueItemsInOrder (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168633:7)
    at processDispatchQueue (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168646:5)
    at dispatchEventsForPlugins (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168657:3)
    at http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168866:12
    ```
<img width="1213" alt="Screenshot 2024-12-11 at 3 46 14 PM" src="https://github.com/user-attachments/assets/1a5e77d0-dc4b-45c5-b9ea-e147e5b81424" />

Reply:
I took a crack at some specs for this and am also struggling to mimic the onBlur behavior for the component. I think the comment helps a lot with the context. Not sure if moving the component out of the input is a valid option or not, but that may cause the blur to not get triggered?

Reply:
UPDATE adding an `event.stopPropagation()` to the Description field fixed this nicely! Got a test in for it, too.